### DISCUSSION TITLE
Cloudflare

### DISCUSSION THREAD

**Discussion by byXav** [LimeDrive/ygg-rss-proxy#2] [General]
Bonjour, 

J'obtiens l'erreur "Authentication failed" à l'utilisation, je n 'arrive pas à trouver pourquoi. 

Le container tourne dans Docker sur une VM Debian sans DE sur TrueNAS (url : http://192.168.1.133/), quand je rentre l'url complète pour obtenir le flux, j'ai l'erreur ci-dessus. 

Url utilisée : http://192.168.1.133:5000/rss?action=generate&type=cat&id=2145&passkey=######

Mon docker-compose.yaml est le suivant : 

`services:`
`  ygg-rss-proxy:`
`    image: ghcr.io/limedrive/ygg-rss-proxy:latest`
`    ports:`
`      - "5000:5000"`
`    environment:`
`      LOCAL_RSS_HOST: 192.168.1.133`
`      LOCAL_RSS_PORT: 5000`
`      YGG_USER: ######`
`      YGG_PASS: ######`

- A tout hasard, j'ai un mot de passe compliqué avec des caractères #(!?*^...  cela pourrait être la cause du soucis.
- J'ai essayé aussi en mettant les valeurs USER et PASS entre guillemets, pas d'amélioration. 
- Tout me semble normal dans Portainer. 

Merci. 

Xavier.

**Comment by LimeDrive** [comment]
Bonjour, retour de la protection Cloudflare sur le site en question.
Les fixe possibles sont beaucoup trop instables pour que je les mette en place sur ce projet.

**Comment by LimeDrive** [comment]
@byXav 

mise en place de flaresolverr pour cloudflare même si ça peut provoquer quelques instabilités.
Attention, j'ai été obligé de changer pas mal de choses dans la configuration.

**Comment by byXav** [comment]
Bonjour, 

Merci pour la MAJ, je vais tester ca. Bonne soirée. 

Xavier.

**Comment by gfarmerfr** [comment]
>     * A tout hasard, j'ai un mot de passe compliqué avec des caractères #(!?*^...  cela pourrait être la cause du soucis.

Salut,

Si tu as des caracteres qui sont interpretés en bash genre $ dans ton password ca va pas marché si tu le colle tel quel.
J'ai testé avec la version flaresolverr. et  il faut mettre le password entre simple quote genre YGG_PASS: 'toto$toto'

**Comment by byXav** [comment]
Salut, 

Cela fonctionne depuis hier soir ! Dḿarrage difficile, je suis resté longtemps sur la page "Authentication failed", avant que cela ne se débloque seul. 
Non pas de $ dans le mot de passe, j'évite à cause de ce genre de soucis (tout comme / et \). 

Merci.

**Comment by LimeDrive** [comment]
Bonjour, pour les mots de passe, tous les caractères doivent être échappés si nécessaire pour fit avec le fichier YAML de Docker Compose.

Si vous avez plusieurs caractères spéciaux à l'intérieur, il faut effectivement mettre le mot de passe dans des guillemets simples.

En résumé :
- Doubler les $ en $$
- Entourer toute la valeur de ' ' si elle contient d'autres caractères spéciaux.

Je vais faire une mise à jour pour l'utilisation des Docker Secrets. Donc avec une config de variables avec FILE_ , ce qui devrai permettre à ceux qui ont des mots de passe complexes de les mettre directement à l'intérieur d'un fichier qu'exploitera Docker sans problème d'échappement.

Concernant le démarrage difficile, la solution actuelle avec flaresolverr restera de toute façon bancale, outre le fait de passer un JS challenge, Cloudflare fait aussi des vérifications au niveau des IP. Il suffit que vous soyez sur une IP de datacenter ou sur un range qui a déjà été flag, pour que le challenge présenté soit un captcha, qui n'est pas résolu par Flaresolverr. Si cela vous arrive sur une IP résidentielle, il suffit d'attendre une douzaine d'heures avant de relancer pour pouvoir être whitelisté en règle général.

**Comment by byXav** [comment]
Cela fonctionne déjà très bien comme ca, merci pour tout ca. Le démarrage a été un peu difficile mais depuis c'est un sans faute. Tu n'as pas un lien pour te tipser des fois comme on peut voir sur d'autres repo du style Ko-fi ou autre ?

**Comment by LimeDrive** [comment]
@byXav Merci de l'intention, Ce projet est simplement un petit passe-temps, donc je ne veux pas le monétiser.

**Demain, je pousserai une version un peu plus aboutie pour sortir de beta.**

Notamment vis-à-vis de la gestion des sessions identifiez sur YGGTorrent. Elles seront désormais partagées entre tous les workers et persistante. 
Le processus d'authentification a aussi été revu.
On pourra désormais vraiment affiner les réglages avec des variables d'environnement supplémentaires. 
FlareSolverr a été inclus autrement qu'avec du bricolage à l'intérieur, donc si nécessité il y a le script l'utilisera sinon il ne l'utilisera pas. 
Il sera désormais possible de faire la configuration avec des dockers secrets. 
Beaucoup de logging a été ajouté ainsi qu'un fichier log.
